exonum/src/runtime/mod.rs

```diff
@@ -14,27 +14,23 @@
 
 use protobuf::well_known_types::Any;
 
-use crate::blockchain::ExecutionError;
 use crate::storage::Fork;
 
+pub mod dispatcher;
+pub mod error;
 mod rust;
 
-#[derive(Debug)]
-pub enum DeployError {
-    WrongArtifact,
-    FailedToDeploy,
-    AlreadyDeployed,
-}
+use error::{DeployError, ExecutionError, InitError};
 
 #[derive(Debug)]
-pub enum InitError {
-    WrongArtifact,
-    NotDeployed,
-    ServiceIdExists,
+pub enum DeployStatus {
+    DeployInProgress,
+    Deployed,
+    DeployErrored(DeployError),
 }
 
 type ServiceInstanceId = u32;
-type InterfaceId = u32;
+type MethodId = String;
```

Comment:
Is this change permanent? Because I think integer method can be useful to minimize space storage + it is quite easy to assign them.

Reply:
Nope, it's temporary. Until we create an entity which will provide handy interface for (name => number) mapping, I think identifiers as strings are a bit more usable.